nixos/modules/services/misc/persistent-evdev.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.persistent-evdev;
+  configFile = pkgs.writeText "persistent-evdev-config.json" ''
+    ${builtins.toJSON {
+      cache = "${cfg.cacheDir}";
+      devices = mapAttrs (virt: phys: "/dev/input/by-id/${phys}") cfg.devices;
+    }}
+  '';
+in
+{
+  options.services.persistent-evdev = {
+    enable = mkEnableOption "virtual input devices that persist even if the backing device is hotplugged";
+
+    cacheDir = mkOption {
```

Comment:
prefer systemd `CacheDirectory` over manually handling these where possible. if the cache has to be moved out of `/var/cache/persistent-evdev` a systemd bind mount can be used.